### PR TITLE
feat(infra): orm f.enum() support

### PR DESCRIPTION
**Pull request by CatsJuice**

### CODE REVIEW

**Comment by CatsJuice** [comment]
* **#9323** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/9323?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/9323?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `canary`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/toeverything?ref=merge-queue-instructions-in-stack-comment&prId=9323).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/frontend/core/src/modules/db/schema/schema.ts [hunk 1]
```diff
@@ -26,7 +26,7 @@ export const AFFiNE_WORKSPACE_DB_SCHEMA = {
     id: f.string().primaryKey().optional().default(nanoid),
     name: f.string().optional(),
     type: f.string(),
-    show: f.string().optional(),
+    show: f.enum('always-show', 'always-hide', 'hide-when-empty').optional(),
```
Comment (forehalo) [inline packages/frontend/core/src/modules/db/schema/schema.ts:29]: if no runtime value check for enum types, the value be written or read may not come to the ones expected

Comment (CatsJuice) [reply packages/frontend/core/src/modules/db/schema/schema.ts:29]: maybe just for type safety is enough?

Comment (forehalo) [reply packages/frontend/core/src/modules/db/schema/schema.ts:29]: if so, you can just simply add a type to `t.string<''>()`, it won't make any difference if there is no runtime value check by `t.enum`

**Review by forehalo — APPROVED** [review]

**Pull request merged**